packages/flutter/lib/src/widgets/platform_menu_bar.dart

```diff
@@ -0,0 +1,975 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+
+import 'binding.dart';
+import 'framework.dart';
+import 'shortcuts.dart';
+
+// "flutter/menu" Method channel methods.
+const String _kMenuSetMethod = 'Menu.setMenu';
+const String _kMenuSelectedCallbackMethod = 'Menu.selectedCallback';
+const String _kMenuItemOpenedMethod = 'Menu.opened';
+const String _kMenuItemClosedMethod = 'Menu.closed';
+
+// Keys for channel communication map.
+const String _kIdKey = 'id';
+const String _kLabelKey = 'label';
+const String _kEnabledKey = 'enabled';
+const String _kChildrenKey = 'children';
+const String _kIsDividerKey = 'isDivider';
+const String _kPlatformDefaultMenuKey = 'platformProvidedMenu';
+
+/// A mixin allowing a [ShortcutActivator] to provide data for serialization of
+/// the shortcut for sending to the platform.
+///
+/// This is meant for those who have written their own [ShortcutActivator]
+/// subclass, and would like to have it work for menus in a [PlatformMenuBar] as
+/// well.
+mixin MenuSerializableShortcut {
```

Comment:
Well, I tried that (it seems like a good idea), but `SingleActivator` only implements `ShortcutActivator`, it doesn't extend it, and apparently that's not sufficient to satisfy "on":

```
'MenuSerializableShortcut' can't be mixed onto 'Object' because 'Object' doesn't implement 'ShortcutActivator'. Try extending the class 'MenuSerializableShortcut'.
```